src/main/scala/Joint.scala

```diff
@@ -49,12 +115,14 @@ object Joint {
   * of the line from the joint's center in the local coordinate system.
   */
 case class Joint(normalVec: (Double, Double, Double), localOrigin: (Double, Double, Double),
-                 center: (Double, Double, Double), dipAngle: Double, dipDirection: Double, phi: Double,
-                 cohesion: Double, shape: Seq[((Double, Double, Double),Double)]) {
+                 center: (Double, Double, Double), phi: Double, cohesion: Double,
+                 shape: Seq[((Double, Double, Double),Double)]) {
   val (a, b, c) = normalVec
   val (centerX, centerY, centerZ) = center
   val d = Joint.findDistance(normalVec, localOrigin, center)
   val (localX, localY, localZ) = localOrigin
+  val dipAngle = Joint.dipAngle(normalVec)
```

Comment:
You know, I just realized that we have a similar problem here to what I had with the bounding sphere. As it is right now, we calculate dip angle and dip direction every time we instantiate a joint object. These calculations involve a bunch of trig stuff, so they're potentially expensive.

Maybe we should use the same solution that I used for the bounding spheres, where there are optional `dipAngle` and `dipDirection` parameters that can be specified so we don't have to recalculate these things. This could also be premature optimization though. What do you think?